backend/validation_results_model.py

```diff
@@ -41,39 +41,49 @@ class ValidationResultsModelBase(Model):
     validation_outcome_index: ValidationOutcomeIdx
 
 
-class ValidationResultsModelMeta(MetaModel):
-    def __new__(
-        cls,
-        name: str,
-        bases: Tuple[Type[object], ...],
-        namespace: Dict[str, Any],
-        discriminator: Optional[Any] = None,
-    ) -> "ValidationResultsModelMeta":
-        namespace["Meta"] = type(
-            "Meta",
-            (),
-            {
-                "table_name": get_param(ParameterName.VALIDATION_RESULTS_TABLE_NAME),
-                "region": environ["AWS_DEFAULT_REGION"],
-            },
-        )
-        klass: "ValidationResultsModelMeta" = MetaModel.__new__(
-            cls, name, bases, namespace, discriminator=discriminator
-        )
-        return klass
-
-
-def validation_results_model_with_meta() -> Type[ValidationResultsModelBase]:
+def validation_results_model_with_meta(
+    results_table_name: Optional[str] = None,
+) -> Type[ValidationResultsModelBase]:
+    if results_table_name is None:
+        results_table_name = get_param(ParameterName.VALIDATION_RESULTS_TABLE_NAME)
+
+    class ValidationResultsModelMeta(MetaModel):
+        def __new__(
+            cls,
+            name: str,
+            bases: Tuple[Type[object], ...],
+            namespace: Dict[str, Any],
+            discriminator: Optional[Any] = None,
+        ) -> "ValidationResultsModelMeta":
+            namespace["Meta"] = type(
+                "Meta",
+                (),
+                {
+                    "table_name": results_table_name,
+                    "region": environ["AWS_DEFAULT_REGION"],
+                },
+            )
+            klass: "ValidationResultsModelMeta" = MetaModel.__new__(
+                cls, name, bases, namespace, discriminator=discriminator
+            )
+            return klass
+
     class ValidationResultsModel(ValidationResultsModelBase, metaclass=ValidationResultsModelMeta):
         validation_outcome_index = ValidationOutcomeIdx()
 
     return ValidationResultsModel
 
 
 class ValidationResultFactory:  # pylint:disable=too-few-public-methods
-    def __init__(self, hash_key: str):
+    def __init__(
+        self,
+        hash_key: str,
+        results_table_name: Optional[str] = None,
```

Comment:
As discussed, we probably want to make this non-optional, to avoid the different code paths in production code.